src/util/makeGraph.ts

```diff
@@ -12,26 +12,24 @@ function makeGetter<GraphType, K extends keyof GraphType>(
   key: K,
 ): () => GraphType[K] {
   return () => {
-    let returnValue: GraphType[K];
-
-    if (components[key] == null) {
-      if (lockedKeys.includes(key)) {
-        const cycle = [...lockedKeys.slice(lockedKeys.indexOf(key)), key].join(
-          " -> ",
-        );
-        throw new Error(`Dependency injection graph cycle detected: ${cycle}`);
-      }
-      const factory = factoryMap[key] as (graph: GraphType) => GraphType[K];
-      lockedKeys.push(key);
-      returnValue = factory(graph);
-      if (lockedKeys.pop() !== key) {
-        throw new Error("Unexpected key at top of graph stack");
-      }
-      components[key] = returnValue;
-    } else {
-      returnValue = components[key] as GraphType[K];
+    const componentValue = components[key];
+    if (componentValue != null) {
+      return componentValue;
```

Comment:
If TypeScript is updated we should be able to still check `components[key]` without an intermediate `const`: https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#control-flow-analysis-for-element-access
```suggestion
    if (components[key] != null) {
      return components[key];
```

Reply:
Nice. Fwiw I'll prob just close this PR tho. I factored it out of another PR, but we're moving to pure DI, so this code is all going to go away